features/dd-sdk-android-trace/src/test/kotlin/com/datadog/opentracing/PendingTraceTest.kt

```diff
@@ -67,6 +81,175 @@ class PendingTraceTest {
         }
     }
 
+    @Test
+    fun `M not leak the PendingTraces W finishSpan`(
+        forge: Forge
+    ) {
+        // Given
+        val mockTracer: DDTracer = mock()
+        val mockInternalLogger: InternalLogger = mock()
+        whenever(mockTracer.partialFlushMinSpans) doReturn 1
+
+        repeat(forge.anInt(min = 10, max = 30)) {
+            val fakeTraceId = BigInteger.valueOf(forge.aLong())
+
+            val pendingTrace = createInstance(
+                PendingTrace::class.java,
+                mockTracer,
+                fakeTraceId,
+                mockInternalLogger
+            )
+
+            val rootSpan =
+                forge.fakeSpan(
+                    pendingTrace,
+                    mockTracer,
+                    fakeTraceId,
+                    BigInteger.ZERO,
+                    forge.anAlphabeticalString(),
+                    mockInternalLogger
+                )
+            Thread.sleep(1)
+            rootSpan.finish()
+        }
+
+        // When
+        PendingTrace.getSpanCleaner().run()
+
+        // Then
+        assertThat(PendingTrace.getPendingTracesSize()).isEqualTo(0)
+    }
+
+    @Test
+    fun `M not leak the PendingTraces W finishSpan from different threads`(
+        forge: Forge
+    ) {
+        // Given
+        val mockTracer: DDTracer = mock()
+        val mockInternalLogger: InternalLogger = mock()
+        whenever(mockTracer.partialFlushMinSpans) doReturn 1
+        val repeatCount = forge.anInt(min = 10, max = 30)
+        val countDownLatch = CountDownLatch(repeatCount)
+        repeat(repeatCount) {
+            Thread {
+                val fakeTraceId = BigInteger.valueOf(forge.aLong())
+
+                val pendingTrace = createInstance(
+                    PendingTrace::class.java,
+                    mockTracer,
+                    fakeTraceId,
+                    mockInternalLogger
+                )
+
+                val rootSpan =
+                    forge.fakeSpan(
+                        pendingTrace,
+                        mockTracer,
+                        fakeTraceId,
+                        BigInteger.ZERO,
+                        forge.anAlphabeticalString(),
+                        mockInternalLogger
+                    )
+                Thread.sleep(1)
+                rootSpan.finish()
+                countDownLatch.countDown()
+            }.start()
+        }
+        countDownLatch.await(20000, TimeUnit.MILLISECONDS)
+
+        // When
+        Thread {
+            PendingTrace.getSpanCleaner().run()
+        }.apply {
+            start()
+            join(20000)
+        }
```

Comment:
we shouldn't have a timeout here, but isn't 20000ms too much for such repetitions count? probably even 1s will be enough.